pypesto/optimize/util.py

```diff
@@ -138,3 +139,38 @@ def bound_n_starts_from_env(n_starts: int):
     )
 
     return n_starts_new
+
+
+def assign_ids(
+    ids: Iterable[str] = None,
+    result: Result = None,
+    n_starts: int = 100,
+) -> Iterable[str]:
+    """
+    Assign ids to starts.
+
+    ids:
+        Ids assigned to the startpoints.
+    result:
+        A result object to append the optimization results to. For example,
+        one might append more runs to a previous optimization. Assign only
+        unique ids.
+    n_starts:
+        Number of starts of the optimizer.
+    """
+    used_ids = set({})
+    if result is not None:
+        used_ids = result.optimize_result.id
```

Comment:
`used_ids` changes to type `list` here (maybe causes the `isdisjoint` call to fail?)
```suggestion
        used_ids = set(result.optimize_result.id)
```
Or change `used_ids.isdisjoint(ids)` to `set(used_ids).isdisjoint(ids)`